scheduler/testdata/testdata.go

```diff
@@ -0,0 +1,96 @@
+package testdata
+
+import (
+	"bufio"
+	_ "embed"
+	"fmt"
+	"log"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	apimeta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiruntime "k8s.io/apimachinery/pkg/runtime"
+	apiyaml "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/kubectl/pkg/scheme"
+	frameworkruntime "k8s.io/kubernetes/pkg/scheduler/framework"
+	"sigs.k8s.io/kube-scheduler-wasm-extension/scheduler/plugin"
+)
+
+// NewPluginExampleFilterSimple returns a new plugin configured with PathExampleFilterSimple.
+func NewPluginExampleFilterSimple() (frameworkruntime.Plugin, error) {
+	return plugin.New(&apiruntime.Unknown{
+		ContentType: "application/json",
+		Raw:         []byte(fmt.Sprintf(`{"guestName":"filter-simple", "guestPath":"%s"}`, PathExampleFilterSimple)),
+	}, nil)
+}
+
+var PathExampleFilterSimple = pathExample("filter-simple")
+
+// NewPluginExampleNoop returns a new plugin configured with PathExampleNoop.
+func NewPluginExampleNoop() (frameworkruntime.Plugin, error) {
+	return plugin.New(&apiruntime.Unknown{
+		ContentType: "application/json",
+		Raw:         []byte(fmt.Sprintf(`{"guestName":"noop", "guestPath":"%s"}`, PathExampleNoop)),
+	}, nil)
+}
+
+var PathExampleNoop = pathExample("noop")
+
+//go:embed testdata/node.yaml
+var yamlNodeReal string
+
+// NodeReal is a realistic v1.Node used for testing and benchmarks.
+var NodeReal = func() *v1.Node {
+	node := v1.Node{}
+	decodeYaml(yamlNodeReal, &node)
+	return &node
+}()
+
+// NodeSmall is the smallest node that works with PathExampleFilterSimple.
+var NodeSmall = &v1.Node{ObjectMeta: apimeta.ObjectMeta{Name: "good-node"}}
+
+//go:embed testdata/pod.yaml
+var yamlPodReal string
+
+// PodReal is a realistic v1.Pod used for testing and benchmarks.
+var PodReal = func() *v1.Pod {
+	pod := v1.Pod{}
+	decodeYaml(yamlPodReal, &pod)
+	return &pod
+}()
+
+// PodSmall is the smallest pod that works with PathExampleFilterSimple.
+var PodSmall = &v1.Pod{Spec: v1.PodSpec{NodeName: NodeSmall.Name}}
```

Comment:
Let's have some metadata which all Pods must have; UID, Pod Name, and Namespace.